### PR TITLE
Add hydrometeor uncertainty

### PR DESCRIPTION
**Pull request by cshartsough**
### DESCRIPTION OF CHANGES:
There is a effort between EMC and GSL to include more information about uncertainty for RTMA aviation-related fields. This is an initial set to include hydrometeor uncertainty in the RTMA output.

Specifically, the following changes were made:

- A previous PR in regional_workflow, #558, uses a new flag, i_uncertainty, to turn this step on/off.  If .true., it creates a template netcdf file, copied from the fcst_fv3lam/INPUT/ fv_tracer file, in the same directory.
- Compute "uncertainty" for the hydrometeors, using the simple formula "unc = 0.1 * (A-B)". This can be updated with more sophisticated calculations at a future time.
- Overwrite the field values in the new file with the uncertainties values.

Note that this is part II of a two-part update. The changes to regional_workflow, which add a flag and create the template netcdf file, are in the regional_workflow code/scripts/exregional_nonvarcldanal.sh script, which were included in a separate PR#558. This PR should be merge *after* or with regional_workflow #PR558.

**Type of change**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x]  New feature (non-breaking change which adds functionality)
- [ ]  Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ]  This change requires a documentation update

### TESTS CONDUCTED:

- [ ]  hera.intel
- [ ]  orion.intel
- [ ]  cheyenne.intel
- [ ]  cheyenne.gnu
- [ ]  gaea.intel
- [x]  jet.intel
- [ ]  wcoss2.intel
- [ ]  NOAA Cloud (indicate which platform)
- [ ]  Jenkins
- [ ]  fundamental test suite
- [ ]  comprehensive tests (specify which if a subset was used)

### DEPENDENCIES:

This is part II of a two-part update. The changes to regional_workflow, which add a flag and create the template netcdf file, are in the regional_workflow code/scripts/exregional_nonvarcldanal.sh script, which were included in a separate PR#558. This PR should be merge *after* or with regional_workflow #PR558.

### DOCUMENTATION:
None

### ISSUE:
None.

### CHECKLIST

- [ ]  My code follows the style guidelines in the Contributor's Guide
- [x]  I have performed a self-review of my own code using the Code Reviewer's Guide
- [x]  I have commented my code, particularly in hard-to-understand areas
- [ ]  My changes need updates to the documentation. I have made corresponding changes to the documentation
- [ ]  My changes do not require updates to the documentation (explain).
- [ ]  My changes generate no new warnings
- [ ]  New and existing tests pass with my changes
- [ ]  Any dependent changes have been merged and published

### CODE REVIEW

File: cloudanalysis/get_fv3sar_bk_mod.f90 [hunk 1]
```diff
@@ -183,15 +195,17 @@ subroutine read_fv3sar_bk
      mype_ql=mype
      mype_2d=mype
 
-     dynvars= bg_fv3regfilenameg%dynvars
-     tracers= bg_fv3regfilenameg%tracers
-     sfcvars= bg_fv3regfilenameg%sfcdata
-     phyvars= bg_fv3regfilenameg%phydata
+     dynvars    = bg_fv3regfilenameg%dynvars
```
Comment (hu5970) [inline cloudanalysis/get_fv3sar_bk_mod.f90]: Any reason to change space around "="?

File: cloudanalysis/cloudanalysis_fv3driver.f90 [hunk 2]
```diff
@@ -1293,6 +1294,14 @@ program cloudanalysis
   do k=1,nsig
      do j=1,lat2
         do i=1,lon2
+           if(i_uncertainty) then
+              ! hydrometeor uncertainties
+              unc_ql(i,j,k) = 0.1 * (cldwater_3d(i,j,k) - ges_ql(i,j,k))
```
Comment (hu5970) [inline cloudanalysis/cloudanalysis_fv3driver.f90]: change 0.1 to 0.1_r_kind

**Comment by cshartsough** [comment]
Thanks for the review so far, I made the changes suggested by Ming and retested to make sure I hadn't missed anything.

**Comment by hu5970** [comment]
@cshartsough This branch is not synced with the develop. Please sync with develop branch first. Thanks, Ming

**Comment by cshartsough** [comment]
This PR was superseded by an updated version, #49    Closing this branch/PR.

**Pull request closed without merging**